### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by MarchD**

### CODE REVIEW

**Review by olesiatk — APPROVED** [review]

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,21 @@
...
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let joinedElements = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        this[i] = '';
```
Comment (aionova) [inline src/arrayMethodJoin.js:16]: You mutate the original array here and you shouldn't

**Review by aionova — CHANGES_REQUESTED** [review]

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const array = [...this];
```
Comment (aionova) [inline src/arrayMethodJoin.js:8]: No need to include data type in the name of a variable. We know that it's an array. 
```suggestion
    const elements = [...this];
```

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,23 @@
...
+      return '';
+    }
+
+    let joinedElements = '';
+
+    for (let i = 0; i < array.length - 1; i++) {
+      if (array[i] === undefined || array[i] === null) {
+        array[i] = '';
+      }
+      joinedElements = joinedElements + array[i] + separator;
```
Comment (aionova) [inline src/arrayMethodJoin.js:20]:
```suggestion
      joinedElements += array[i] + separator;
```

**Review by aionova — APPROVED** [review]
Good job!

**Pull request closed without merging**